test/amqp-integration-test.py

```diff
@@ -36,7 +36,7 @@ def __init__(self, cmd, proc):
 def run(path):
     global processes
     binary = os.path.join(tempdir, os.path.basename(path))
-    cmd = 'go build -tags pkcs11 -o %s %s' % (binary, path)
+    cmd = 'GORACE="halt_on_error=1" go build -tags pkcs11 -race -o %s %s' % (binary, path)
```

Comment:
I considered that, but I didn't want to infect all the other commands from a distance. If someone adds more go code that shouldn't be exit under a race detector failure or should but it runs before this code is called, they will get behavior they don't expect. Action at a distance here is not super helpful long-term